src/authn/ccloudStateHandling.ts

```diff
@@ -1,284 +1,153 @@
-        await getCCloudAuthSession(true);
+        logger.debug(`CCloud connection in initial ${ConnectedState.None} state (no tokens)`);
       }
-    });
-}
+      break;
 
-/** Check the {@link Connection} for any auth-related errors and prompt the user to reauthenticate. */
-export function checkAuthErrors(connection: Connection) {
-  const errors: AuthErrors | undefined = connection.status.authentication.errors;
-  if (!errors) {
-    return;
+    default:
+      logger.warn(`handleUpdatedConnection: unhandled CCloud state ${connectedState}`);
   }
 
-  // tell the auth provider that the session is invalid so it can prompt the user to log in again
-  // and clear out the sidebar resources (but not delete the connection)
-  ccloudAuthSessionInvalidated.fire();
-
-  const tracker = AuthPromptTracker.getInstance();
-  logger.error("errors returned while checking auth status", {
-    connectionId: connection.spec.id,
-    errors,
-    promptingUser: !tracker.authErrorPromptOpen,
-  });
-
-  if (tracker.authErrorPromptOpen) {
-    // if we've already prompted the user to reauthenticate, don't do it again unless they dismiss
-    // it and we continue to see errors
-    return;
-  }
+  const errors: AuthErrors | undefined = status.errors;
+  if (errors && Object.keys(errors).length > 0) {
+    logger.debug("checking CCloud status errors", { errors });
 
-  // show an error message to the user to retry the auth flow
-  tracker.authErrorPromptOpen = true;
-  vscode.window
-    .showErrorMessage(
-      "Error authenticating with Confluent Cloud. Please try again.",
-      CCLOUD_SIGN_IN_BUTTON_LABEL,
-    )
-    .then(async (response: string | undefined) => {
-      // always reset the prompt tracker after the user interacts with the notification in any way,
-      // since they will either dismiss it (and we re-prompt at the next iteration) or they re-
-      // authenticate and we get a new status back (or another auth error at the next iteration)
-      tracker.authErrorPromptOpen = false;
-      if (response === CCLOUD_SIGN_IN_BUTTON_LABEL) {
-        // if we got to this point, we likely cleared out the existing connection via the
-        // `ccloudAuthSessionInvalidated` emitter, so we need to create a new session to re-auth
-        await getCCloudAuthSession(true);
+    // parse AuthErrors to see if we have any transient (or non-transient) errors
+    const isTransientValues: boolean[] = [];
+    Object.values(errors).forEach((error: AuthError) => {
+      const isTransient: boolean | undefined = error.is_transient;
+      if (isTransient !== undefined) {
+        isTransientValues.push(isTransient);
       }
     });
-}
-
-export function openExternal(uri: vscode.Uri) {
-  if (process.env.NODE_ENV === "testing") {
-    // XXX never remove this log, tests rely on it
-    logger.info("actionOpenExternal", uri.toString());
-    return Promise.resolve(true);
```

Comment:
This hasn't actually been used since before this repo went public. Back in the early prototype days, this was used for a test that was deleted back in PR 463 (in the old repo). Nothing uses it now.